common/src/Common/Wallet.hs

```diff
@@ -678,6 +680,17 @@ accountUnfinishedCrossChainTransfer = view (accountInfo . accountInfo_unfinished
 accountBalance :: Account -> Maybe AccountBalance
 accountBalance = view (accountInfo . accountInfo_balance)
 
+upsertNonVanityBalance :: NetworkName -> PublicKey -> ChainId -> Maybe AccountBalance -> AccountStorage -> AccountStorage
+upsertNonVanityBalance net pk chain balance store = store
+  & (_AccountStorage . at net %~ (Just . maybe mempty id))
+  & (_AccountStorage . ix net . accounts_nonVanity . at pk %~ (Just . maybe mempty id))
```

Comment:
`(Just . fold)` instead of `(Just . maybe mempty id)` ?

Reply:
Oh yeah, good point. :)